src/test/scala/com/databricks/spark/redshift/RedshiftSuite.scala

```diff
@@ -0,0 +1,458 @@
+import java.util.Properties
+
+import org.apache.spark.SparkContext
+import org.apache.spark.sql.jdbc.DefaultJDBCWrapper
+import org.apache.spark.sql.{Row, SQLContext, SaveMode}
+import org.scalamock.scalatest.MockFactory
+import org.scalatest.{BeforeAndAfterAll, FunSuite, Matchers}
+
+/**
+ * Tests main DataFrame loading and writing functionality
+ */
+class RedshiftSuite extends FunSuite with Matchers with MockFactory with BeforeAndAfterAll {
+  val jdbcUrl = s"${System.getenv("aws_redshift_jdbc_url")}?" +
+    s"user=${System.getenv("aws_redshift_user")}&password=${System.getenv("aws_redshift_password")}"
+  val tempDir = System.getenv("aws_s3_scratch_space")
+
+  val accessKeyId = System.getenv("aws_access_key_id")
+  val secretAccessKey = System.getenv("aws_secret_access_key")
+
+  /**
+   * Expected parsed output corresponding to the output of testData.
+   */
+  val expectedData = Array(
+    Row(List.fill(10)(null): _*),
+    Row(0.toByte, null, TestUtils.toDate(2015, 6, 3), 0.0, -1.0f, 4141214, 1239012341823719L, null, "f",
+      TestUtils.toTimestamp(2015, 6, 3, 0, 0, 0)),
+    Row(0.toByte, false, null, -1234152.12312498, 100000.0f, null, 1239012341823719L, 24.toShort, "___|_123", null),
+    Row(1.toByte, false, TestUtils.toDate(2015, 6, 2), 0.0, 0.0f, 42, 1239012341823719L, -13.toShort, "asdf",
+      TestUtils.toTimestamp(2015, 6, 2, 0, 0, 0, 0)),
+    Row(1.toByte, true, TestUtils.toDate(2015, 6, 1), 1234152.12312498,
+      1.0f, 42, 1239012341823719L, 23.toShort, "Unicode's樂趣", TestUtils.toTimestamp(2015, 6, 1, 0, 0, 0, 1))
+  )
+
+
+  /**
+   * Spark Context with hadoop file overridden to point at our local test data file for this suite,
+   * no-matter what temp directory was generated and requested.
+   */
+  private var sc: SparkContext = _
+  private var conn: Connection = _
+
+  override def beforeAll(): Unit = {
+    super.beforeAll()
+    sc = new SparkContext("local", "RedshiftSourceSuite")
+
+    sc.hadoopConfiguration.set("fs.s3n.awsAccessKeyId", accessKeyId)
+    sc.hadoopConfiguration.set("fs.s3n.awsSecretAccessKey", secretAccessKey)
+
+    conn = DefaultJDBCWrapper.getConnector("org.postgresql.Driver", jdbcUrl, new Properties())()
+
+    conn.prepareStatement("drop table if exists test_table").executeUpdate()
+    conn.prepareStatement("drop table if exists test_table2").executeUpdate()
+    conn.prepareStatement("drop table if exists test_table3").executeUpdate()
+    conn.commit()
+
+    def createTable(tableName: String): Unit = {
+      conn.prepareStatement(
+        s"""
+           |create table $tableName (
+           |testbyte int2,
+           |testbool boolean,
+           |testdate date,
+           |testdouble float8,
+           |testfloat float4,
+           |testint int4,
+           |testlong int8,
+           |testshort int2,
+           |teststring varchar(256),
+           |testtimestamp timestamp
+           |)
+      """.stripMargin
+      ).executeUpdate()
+
+      conn.prepareStatement(
```

Comment:
Why create a separate prepared statement for each row instead of creating a single parameterized prepared statement?